### DISCUSSION TITLE
NAudio 2.2.1

### DISCUSSION THREAD

**Discussion by galleonms** [naudio/NAudio#1159] [Q&A] [answered]
Hi @markheath ,

We recently made an update to our WPF application and increase the NAudio version from 2.0.0, to 2.2.1. Our application is for both Windows and Mac (code wrapped).

We use NAudio to calculate BPM and display energy graphs (wave forms) for music files (*.mp3, *.avi, *.mp4, *.mpg, *.wma ...). On Windows devices everything is working as it should but on MacOS since we update the NUGET package it is not working anymore (for version 2.0.0, we could generate energy graphs and calculate BPM only for *.mp3).

Is it so that NAudio is not supported on MacOS anymore. Is there a workaround that we could use?

We don't want to downgrade to 2.0.0 since it is not a good idea.

Waiting for your reply. 

Best regards,
Paul

**Comment by Axwabo** [comment]
Check out [NLayer](https://github.com/naudio/NLayer) for mp3 decoding

**Comment by markheath** [comment, marked as answer]
NAudio has some parts which are cross-platform, but the majority of APIs it provides wrappers for are Windows-only. 
As axwabo said, NLayer will support mp3 decoding.
With NAudio 3 (currently in preview), there is now support for libsndfile, in the NAudio.SoundFile assembly. This should let you use a variety of file types on MacOS. There is some [documentation here](https://github.com/naudio/NAudio/blob/main/Docs/CrossPlatformAudioFilesWithSoundFile.md)
